### ISSUE TITLE
String failure in latest in "Remove swap entries"

### ISSUE THREAD

**Issue by jberkus** [openshift/openshift-ansible#4025]
#### Description

Provide a brief description of your issue here. For example:

Platform:

Fedora 25
O-A HEAD from April 25
Ansible 2.2.1.0

Running O-A against Fedora Atomic Host 25, I get a fatal error in this module:

```
TASK [openshift_node : Remove swap entries from /etc/fstab] ********************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: cannot use a string pattern on a bytes-like object
fatal: [node1.projectatomic.rocks]: FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_s7rnzix6/ansible_module_replace.py\", line 180, in <module>\n    main()\n  File \"/tmp/ansible_s7rnzix6/ansible_module_replace.py\", line 153, in main\n    result = re.subn(mre, params['replace'], contents, 0)\n  File \"/usr/lib64/python3.5/re.py\", line 193, in subn\n    return _compile(pattern, flags).subn(repl, string, count)\nTypeError: cannot use a string pattern on a bytes-like object\n",
    "module_stdout": ""
}

Questions: this looks like a straight-up bug in string manipulation, possibly in ansible itself.   Ideas?

Also, why are we disabling swap?  That seems sub-optimal.

**Comment by jberkus** [comment]
Note that, if I try Ansible 2.3.3.0, I get this issue: #4022

**Comment by smarterclayton** [comment]
We disable swap because Kubernetes can't make effective decisions about memory if the OS lies to it.  Kubernetes is managing overcommit, not the OS.  See long discussions in mailing lists and OpenShift docs on this.

**Comment by jberkus** [comment]
@smarterclayton ah, this explains some of the capacity issues I'm having on the microcluster.  Given that I have a stack of boards with only 2GB ram each but with SSDs, I'd rather have the swap.  But I'm  a very special case, so I'll make my own modifications.

**Comment by sdodson** [comment]
https://github.com/openshift/openshift-ansible/pull/4017 will allow you to leave swap on once it's merged

**Comment by mtnbikenc** [comment]
This appears to be a Python 3 compatibility issue fixed in Ansible 2.3. https://github.com/ansible/ansible/issues/20326

**Comment by jberkus** [comment]
OK, closing this in favor of #4022
